shared/pubsub.js

```diff
@@ -0,0 +1,386 @@
+'use strict'
+
+import type { JSONObject, JSONType } from '~/shared/types.js'
+
+// ====== Types ====== //
+
+export type Message = {
+  [key: string]: JSONType,
+  +type: string
+}
+
+export type PubSubClient = {
+  connectionTimeoutID: TimeoutID | void,
```

Comment:
Where does this `TimeoutID` come from? Similarly, can a comment be added explaining?

Reply:
It's also a Flow thing, more precisely, an opaque type declared in its core definition file:
```js
declare opaque type TimeoutID;
```

Reply:
On the first instance of `TimeoutID` usage in the code, could you similarly add a comment about this? It will help save future developers or archeologists a lot of time when browsing through the code. :)

E.g. something along these lines (if `connectionTimeoutID` is the first place where `TimeoutID` appears):

```js
  connectionTimeoutID: TimeoutID | void, // TimeoutID is an opaque type declared in Flow's core definition file
```

Reply:
Since there is going to be several Flowtype usage related comments, what about grouping them near the top of the file (before the type definitions), like:
```js
/*
 * Flowtype usage notes:
 *
 * - The '+' prefix indicates properties that should not be re-assigned or
 *   deleted after their initialization.
 *
 * - 'TimeoutID' is a predefined Flow core type, used as the return type of the
 *   core setTimeout() function.
 */
 ```
 Of course feel free to suggest improvements!

Reply:
That sounds like a great idea, go for it! 👍